lib/trento_web/controllers/v1/settings_controller.ex

```diff
@@ -144,4 +153,6 @@ defmodule TrentoWeb.V1.SettingsController do
         {:error, :not_found}
     end
   end
+
+  def get_policy_resource(_), do: Trento.Settings.ApiKeySettings
```

Comment:
Settings are based on different readmodels this should change based on the action, you have conn as first parameter, you can pattern match and provide the right readmodel to the policy based on the action

Reply:
Most probably you should pattern match with `body_pararms`.
Something like:
def get_policy_resource(%{body_params: %ApiKeySettingsUpdateRequest{}}), do: Trento.Settings.ApiKeySettings

def get_policy_resource(%{body_params: %ActivityLogSettings{}}), do: Trento.ActivityLog.Settings

Reply:
You cannot, it is used by the policy at the end, which is in other module